arangod/Cluster/Maintenance.cpp

```diff
@@ -848,36 +873,40 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
         auto const planPath = std::vector<std::string>{dbName, colName, "shards", shName};
         if (!pdbs.hasKey(planPath)) {
           LOG_TOPIC("43242", DEBUG, Logger::MAINTENANCE)
-            << "Ooops, we have a shard for which we believe to be the leader,"
-               " but the Plan does not have it any more, we do not report in "
-               "Current about this, database: " << dbName
-            << ", shard: " << shName;
+              << "Ooops, we have a shard for which we believe to be the leader,"
+                 " but the Plan does not have it any more, we do not report in "
+                 "Current about this, database: "
+              << dbName << ", shard: " << shName;
           continue;
         }
 
         VPackSlice thePlanList = pdbs.get(planPath);
         if (!thePlanList.isArray() || thePlanList.length() == 0 ||
-            !thePlanList[0].isString() ||
-            !thePlanList[0].isEqualStringUnchecked(serverId)) {
+            !thePlanList[0].isString() || !thePlanList[0].isEqualStringUnchecked(serverId)) {
           LOG_TOPIC("87776", DEBUG, Logger::MAINTENANCE)
-            << "Ooops, we have a shard for which we believe to be the leader,"
-               " but the Plan says otherwise, we do not report in Current "
-               "about this, database: " << dbName << ", shard: " << shName;
+              << "Ooops, we have a shard for which we believe to be the leader,"
+                 " but the Plan says otherwise, we do not report in Current "
+                 "about this, database: "
+              << dbName << ", shard: " << shName;
           continue;
         }
 
+        auto lastKnownShardsInCurrentPath =
+            std::vector<std::string>{COLLECTIONS, dbName, colName, shName, SERVERS};
+        auto lastKnownShards = cur.get(lastKnownShardsInCurrentPath);
```

Comment:
I find `lastKnownShards` a little unfortunately named, as it's actually a list of servers.